docs/development/npm-package-development.md

```diff
@@ -0,0 +1,152 @@
+
+* **engines** - This is recommended to show which versions of Node.js your package supports.  If this is a public module, it is recommended to try and support the latest LTS versions.  If this is a private package, then this value could be determined by your company.
+
+* **private** - This field is only recommended and should be set to `yes` if your package will not be published to npm.
+
+* **support** - This field is to help package maintainers communicate with and set expectations for their users about the level of support they are willing to provide on a package.  The team has worked closely with the [Node.js Package Maintence Team](https://github.com/nodejs/package-maintenance) on their recommendations on what this support field should look like.  For one of your modules, [opossum](https://github.com/nodeshift/opossum), we [set this field](https://github.com/nodeshift/opossum/blob/main/package.json#L75) to `true` and supplied our support information in a separate [package-support.json file](https://github.com/nodeshift/opossum/blob/main/package-support.json).
+
+  For more information on the support field, check out what the [Package Maintence Team says](https://github.com/nodejs/package-maintenance/blob/main/docs/PACKAGE-SUPPORT.md)
+
+* **type** - This field defines the module format that Node.js will use.  For ES Modules(ESM), use `module` and for Common JS(CJS) modules, use `commonjs`.  For more information on how Node.js determines the difference between ESM and CJS modules take a [look at the Node.js official docs here](https://nodejs.org/dist/latest-v18.x/docs/api/packages.html#determining-module-system)
+
+
+#### Development Dependencies
+
+While there is no list of development dependencies that the team recommends, we do find that it is helpful to agree on the same list across a team of develoeprs.  This is especially important for linting and testing libraries.  Having a similar set can help with ramp up time if a developer moves teams.
+
+#### Dependencies
+
+For the teams guidance on choosing dependencies, check out the [Choosing and vetting dependencies](./dependencies).
+
+* add something about best practice with semver things
+
+During development of a pacakge, it is helpful to add the package-lock.json to source control, so all developers working on the package will be able to install the same versions of dependencies.
+
+It is also recommended to know what semver range you are specifying for any dependecy you wish to install.  For example:
+
+The `^` will give you all the minor and patch releases when they become available.
+The `~` will include everything greater than a particular version in the same minor range.
+
+This [semver calculator](https://semver.npmjs.com/) is a great resource to make this determination
+
+### CJS / ESM
+
+TBD
```

Comment:
- export maps
- combined esm/cjs packages
- prefer esm source with cjs generated by build tools
why? interop (browser, deno), standards

common pitfalls:
-  `__dirname` vs `import.meta.url`
- top level await
- workarounds: do stuff in async iife's (`await import('mymodule');`)